dask/dataframe/core.py

```diff
@@ -780,6 +780,45 @@ def min(self, axis=None, skipna=True):
             return self._aca_agg(token='min', func=_min,
                                  skipna=skipna, axis=axis)
 
+    def _idxmaxmin(self, fn, axis=None, skipna=True):
+        if fn == 'idxmax':
+            idxfn = _idxmax
+        elif fn == 'idxmin':
+            idxfn = _idxmin
+        else:
+            raise ValueError('Function must be idxmax or idxmin')
+
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            return map_partitions(idxfn, None, self,
+                                  token=self._token_prefix + fn,
+                                  skipna=skipna, axis=axis)
+        else:
+            def chunk(x):
+                idx = getattr(x, fn)()
+                value = x.iloc[idx]
+                n = len(x)
+                return pd.DataFrame([[idx, value, n]], columns=['idx','value', 'n'])
+
+            def agg(x, **kwargs):
+                print(x)
+                idx = getattr(x.value, fn)()
+                if self.known_divisions == True:
+                    return x.idx.iloc[idx]
```

Comment:
I recommend pulling them out of the class altogether.  These functions don't require any knowledge of dask.dataframes, so they probably shouldn't be attached to the dask.dataframe classes.  There are a lot of other similar small functions floating around in core.py.